holoviews/core/util/types.py

```diff
@@ -0,0 +1,116 @@
+import datetime as dt
+import inspect
+from types import GeneratorType
+from typing import TYPE_CHECKING
+
+from .dependencies import _lazy_module
+
+if TYPE_CHECKING:
+    import cftime
+    import numpy as np
+    import pandas as pd
+else:
+    cftime = _lazy_module("cftime", bool_use_sys_modules=True)
+    np = _lazy_module("numpy", bool_use_sys_modules=True)
+    pd = _lazy_module("pandas", bool_use_sys_modules=True)
+
+
+# gen_types is copied from param, can be removed when
+# we support 2.2 or greater
```

Comment:
Before doing that, I think `gen_types` should be documented/officially supported in Param. As far as I can see this wasn't done in https://github.com/holoviz/param/pull/966/files.

Reply:
I think this is a very library-related need; if you have an app, you are likely to do an import of the module and use `isinstance` or `issubclass` as you always have. 

I think we should document it, but I want to be very careful about it, as it could be more confusing than helpful.